Add save_excel and save_csv methods to FitData object

OhadNir9 opened the pull request:
Solves #52 and #53

saroad2 commented on src/eddington/fit_data.py:
```diff
@@ -436,3 +436,67 @@ def __is_number(cls, string):
         except ValueError:
             return False
+
+    def __make_sheet(self, sheet=None):
```
There is no separate logic here that may be called in more than one place.

You can move the entire section to the `save_excel` method

saroad2 commented on src/eddington/fit_data.py:
```diff
@@ -436,3 +436,67 @@ def __is_number(cls, string):
+        """
+        Save :class:`FitData` to xlsx file.
+
+        :param output_directory: str or Path.
```
Specify the type of the parameter in a `:type:` section.

saroad2 commented on src/eddington/fit_data.py:
```diff
@@ -436,3 +436,67 @@ def __is_number(cls, string):
+         Path to the directory for the new excel file to be saved.
+        :param name: Optional. The name of the file, without the .csv suffix
+        """
+        worksheet = self.__make_sheet()[1]
```
There is no need to make a worksheet here, especially if you drop it emidiatly after.

I would recommend you use [DictWriter](https://docs.python.org/3/library/csv.html#csv.DictWriter) in order to write the csv file.

saroad2 commented on src/eddington/fit_result.py:
```diff
@@ -9,7 +9,7 @@
 
 @dataclass(repr=False)  # pylint: disable=too-many-instance-attributes
-class FitResult:
+class FitResult:  # pylint: disable=too-many-instance-attributes
```
This fix is not relevant to this new feature and it should be in a different PR

saroad2 commented on tests/fit_data/test_save_data_to_file.py:
```diff
@@ -0,0 +1,77 @@
+from pathlib import Path
```
Those are actually very solid **integration tests**.

One key element of **unit tests** is that they don't interact with the operating system (aka, read and write files).
*Integration tests* however, do exactly that. They test the software end-to-end and see that the different components combine nicely together.

Please keep those tests as they are, but move them to the *integration_tests* directory (if it's missing in your branch, update the master branch from upstream, and merge it into your feature branch. I added this directory yesterday).

After you do that, write new unit tests, but use [mocks](https://realpython.com/python-mock-library/) in order to simulate writing to a file.

Mocks can sometimes be perceived as a little bit hard concept for developers who are not familiar with them. Please feel free to say that you rather I will write the unit tests for you, and I'll step right in.

saroad2 reviewed (changes requested):
Thank you for your contribution.

I left some comments. Please feel free to return a feedback or ask for help if needed.

saroad2 commented on src/eddington/fit_data.py:
```diff
@@ -436,3 +436,67 @@ def __is_number(cls, string):
+
+        workbook.save(path)
+
+    def save_csv(self, output_directory, name=None):
```
use default `name="fit_data.csv"`

saroad2 commented on src/eddington/fit_data.py:
```diff
@@ -436,3 +436,67 @@ def __is_number(cls, string):
+        if name:
+            path = Path(output_directory / f"{name}.csv")
+        else:
+            path = Path(output_directory / "FitData.csv")
```
Remove this if-else statement and use default value in the method signature

saroad2 commented on src/eddington/fit_data.py:
```diff
@@ -436,3 +436,67 @@ def __is_number(cls, string):
+
+        return workbook, worksheet
+
+    def save_excel(self, output_directory, name=None, sheet=None):
```
use `name="fit_data.xlsx"`

saroad2 commented on src/eddington/fit_data.py:
```diff
@@ -436,3 +436,67 @@ def __is_number(cls, string):
+        if name:
+            path = Path(output_directory / f"{name}.xlsx")
+        else:
+            path = Path(output_directory / "FitData.xlsx")
```
Remove if-else statement and use method signature default instead.

OhadNir9 commented:
I'm having trouble with the mock concept, if it's ok I prefer you'll do it. I moved the test file to integration_tests.
Moreover, as you requested I removed the pylint comment from fit_result.

saroad2 reviewed (approved):
Thank you for your contribution!

I added the missing unit tests and fixed some small issues.
